### PR TITLE
Add file streaming examples

### LINKED ISSUE #133: Add FS / HTTP streaming examples
Some questions come up on the community Discord on a regular basis, that we should probably add examples for in the manual:

- How do I stream a response from `fetch` to a file on disk?
- How do I stream a file on disk to a response for `Deno.serveHttp`?
- How do I upload stream a file to a web service via fetch?

### PR DESCRIPTION
**Pull request by tommy-josepovic**
Closes #133 

Changes:
- Add a section about files and streaming to the [Fetch Data example](https://deno.land/manual@v1.16.0/examples/fetch_data).
- Rewrote the [File Server example](https://deno.land/manual@v1.16.0/examples/file_server) to explains how people can write their own file server from scratch. 

I tried to keep the file server example as simple as possible, but it may be missing some exception handling. Should I add it?

### CODE REVIEW

File: examples/fetch_data.md [hunk 1]
```diff
@@ -50,3 +50,40 @@ try {
...
 ```
+
+## Files and Streams
+
+Like in browsers, sending and receiving large files is possible thanks to the
+[Streams API](https://developer.mozilla.org/en-US/docs/Web/API/Streams_API). The
+standard library's [streams module](https://deno.land/std@$STD_VERSION/streams/)
+can be used to convert a Deno file into a writable or readable stream.
+
+**Command:** `deno run --allow-read --allow-write --allow-net fetch-file.ts`
```
Comment (bartlomieju) [inline examples/fetch_data.md:61]: Prefer to use underscore for file names
```suggestion
**Command:** `deno run --allow-read --allow-write --allow-net fetch_file.ts`
```

File: examples/fetch_data.md [hunk 2]
```diff
@@ -50,3 +50,40 @@ try {
...
+
+```tsx
```
Comment (bartlomieju) [inline examples/fetch_data.md:63]:
```suggestion
```ts
```

File: examples/file_server.md [hunk 3]
```diff
@@ -2,62 +2,61 @@
...
-    -p, --port <PORT>   Set port
-    --cors              Enable CORS via the "Access-Control-Allow-Origin" header
-    --host     <HOST>   Hostname (default is 0.0.0.0)
-    -c, --cert <FILE>   TLS certificate file (enables TLS)
-    -k, --key  <FILE>   TLS key file (enables TLS)
-    --no-dir-listing    Disable directory listing
+## Example
 
-    All TLS options are required when one is provided.
+**Command:** `deno run --allow-read --allow-net file-server.ts`
```
Comment (bartlomieju) [inline examples/file_server.md:22]:
```suggestion
**Command:** `deno run --allow-read --allow-net file_server.ts`
```

**Review by bartlomieju — COMMENTED** [review]
@tommy-josepovic looks great! I have a few nitpick comments. Also I think it would be worth to mention existence of `file_server` from the `deno_std` in the example page - ie. your work on showing how to write it from scrach is great, but it'd be nice to add another paragraph below pointing to existing implementation

**Comment by tommy-josepovic** [comment]
I have modified my PR with your recommendations.
I restored most of the original text for the `std/http` file server.

**Comment by bartlomieju** [comment]
> Also, small detail, but Deno.open doesn't work on Deno Deploy. It makes sense since GitHub isn't appropriate for storing large files, but it does make Deno Deploy less compatible with non-Deno Deploy code 😕

This is expected - not all `Deno` namespace APIs are available on Deploy

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @tommy-josepovic

**Comment by lucacasonato** [comment]
> Also, small detail, but Deno.open doesn't work on Deno Deploy. It makes sense since GitHub isn't appropriate for storing large files, but it does make Deno Deploy less compatible with non-Deno Deploy code confused

We are working on it!

File: examples/file_server.md [hunk 4]
```diff
@@ -2,17 +2,74 @@
...
+```ts
+import { readableStreamFromReader } from "https://deno.land/std@$STD_VERSION/streams/mod.ts";
+
+// Start listening on port 8080 of localhost.
+const server = await Deno.listen({ port: 8080 });
+console.log("File server running on http://localhost:8080/");
+
+for await (const conn of server) {
+  const httpConn = Deno.serveHttp(conn);
+  for await (const requestEvent of httpConn) {
```
Comment (sant123) [inline examples/file_server.md:33]: Please see https://deno.land/manual@v1.16.1/runtime/http_server_apis#serving-http

This example will hang the server once a connection is established.

**Review by sant123 — COMMENTED** [review]
Could this be fixed in a PR?

File: examples/file_server.md (hunk 4 above)
Comment (bartlomieju) [reply examples/file_server.md:33]: Good catch; this bit should be reworked then. @sant123 PRs are welcome!

Comment (sant123) [reply examples/file_server.md:33]: Just created https://github.com/denoland/manual/pull/160

**Pull request merged**